### PR TITLE
Only mark the HEAD of PR pending if has fixup commits

### PR DESCRIPTION
**Pull request by deiwin**
This fixes an issue where if a new commit is added on top of fixup
commits that are marked as pending, github still thinks that things are
fine, as the last commits statuses are the ones that are taken into
account for the overall PR status.

We don't have to worry about cleaning up these pending statuses, because
they will always be either on or after fixup and squash commits. This
means that after `!squash` they will get new SHAs and therefore the
statuses will be forgotten.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -150,20 +150,32 @@ func handlePullRequest(w http.ResponseWriter, body []byte, git Git, githubClient
...
+			message := fmt.Sprintf("Failed to create a pending status for commit %s", *pr.Head.SHA)
+			return ErrorResponse{err, http.StatusBadGateway, message}
+		}
+	}
+	return SuccessResponse{}
+}
+
+func includeFixupCommits(commits []github.RepositoryCommit) bool {
 	for _, commit := range commits {
 		if strings.HasPrefix(*commit.Commit.Message, "fixup! ") || strings.HasPrefix(*commit.Commit.Message, "squash! ") {
```
Comment (stenver) [inline main.go]: downcase?

**Comment by stenver** [comment]
+1

File: main.go (hunk 1 above)
Comment (deiwin) [inline main.go]: I don't think git supports commits starting with, say `FIXUP! `. I don't think we should either

**Comment by chris-salemove** [comment]
Can you limit line width to 100 or 120?

**Comment by chris-salemove** [comment]
Is there a good way to test this? I know this is a helper bot - not prod code, but if it will be mucking with our github data and we are sharing it with the world, we test.

**Comment by deiwin** [comment]
Regarding tests, there's 2 other PRs, one of them ads tests for the main flow. I would also like to test the parts that directly deal with git, but I haven't thought of a good way to do that yet.

**Comment by chris-salemove** [comment]
Could test against a throwaway repo. Delete after tests.

**Comment by deiwin** [comment]
Regarding line lengths and making code more granular, I will keep larger formatting/stylistic changes until after the current PRs are merged, because I've built quite a lot on top of these changes and the merge conflicts would be a major pain that I don't want to deal with.

**Comment by deiwin** [comment]
Although I agree that anything above 120 chars in width becomes difficult to review in GitHub.

File: main.go [hunk 2]
```diff
@@ -150,20 +150,32 @@ func handlePullRequest(w http.ResponseWriter, body []byte, git Git, githubClient
...
 	}
+	if includesFixupCommits(commits) {
+		pr, _, err := githubClient.PullRequests.Get(pullRequestEvent.Repository.Owner, pullRequestEvent.Repository.Name, pullRequestEvent.IssueNumber)
+		if err != nil {
+			message := fmt.Sprintf("Getting PR %s/%s#%d failed", pullRequestEvent.Repository.Owner, pullRequestEvent.Repository.Name, pullRequestEvent.IssueNumber)
+			return ErrorResponse{err, http.StatusBadGateway, message}
+		}
+		_, _, err = githubClient.Repositories.CreateStatus(pullRequestEvent.Repository.Owner, pullRequestEvent.Repository.Name, *pr.Head.SHA, &github.RepoStatus{
+			State:       github.String("pending"),
+			Description: github.String("This PR contains commits that need to be squashed with !squash before merging"),
```
Comment (indrekj) [inline main.go]: can we shorten this text a bit? 
screen shot 2015-10-13 at 19 31 53

Comment (deiwin) [inline main.go]: `This PR needs to be squashed with !squash before merging`?

Comment (stenver) [inline main.go]: `!squash this PR before merging` is also an alternative

Comment (indrekj) [inline main.go]: Squish it, smush it, squash it, smash it

Comment (deiwin) [inline main.go]: Will go with my version atm, because it's more of a description of a status and less of a command that way

**Comment by indrekj** [comment]
+1 from me, I'll let @chris-salemove have one more look

**Comment by chris-salemove** [comment]
!squash

**Pull request merged**